subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CachedTaskExecutionIntegrationTest.groovy

```diff
@@ -485,4 +485,34 @@ class CachedTaskExecutionIntegrationTest extends AbstractIntegrationSpec impleme
         then:
         result.assertTaskSkipped(":cacheable")
     }
+
+    def "downstream task stays cached when upstream task is loaded from cache"() {
```

Comment:
Yes. If we store different stuff in the FS mirror while unpacking than we did during the previous run when the task was actually executed, then we'll get the downstream task out-of-date.